extensions/ql-vscode/src/common/vscode/telemetry.ts

```diff
@@ -271,21 +158,45 @@ export class ExtensionTelemetryListener
   set cliVersion(version: SemVer | undefined) {
     this.cliVersionStr = version ? version.toString() : NOT_SET_CLI_VERSION;
   }
+}
 
-  private disposeReporter() {
-    if (this.reporter) {
-      void this.reporter.dispose();
-      this.reporter = undefined;
+async function notifyTelemetryChange() {
+  const continueItem = { title: "Continue", isCloseAffordance: false };
+  const vsCodeTelemetryItem = {
+    title: "More Information about VS Code Telemetry",
+    isCloseAffordance: false,
+  };
+  const codeqlTelemetryItem = {
+    title: "More Information about CodeQL Telemetry",
+    isCloseAffordance: false,
+  };
+  let chosenItem;
+
+  do {
+    chosenItem = await window.showInformationMessage(
```

Comment:
It's slightly annoying you get a "cancel" button but if you click it the dialog just comes back. Sadly I can't see a way to get rid of it and there's nothing sensible we can really do to "cancel" this update, so I agree I think we just leave it like this. Users will only see it once at least.

![Screenshot 2024-12-05 at 16 41 03](https://github.com/user-attachments/assets/835b17a4-45eb-41d5-b990-88215d89b332)